<launch>
  <arg name="stream_rate" default="100"/>
  <arg name="logging_start_delay" default="5" />
  <arg name="do_logging" default="false" />

  <node pkg="mavros" type="mavros_node" name="mavros" required="true" clear_params="true" output="screen" respawn="false"/>
  <node pkg="rosservice" type="rosservice" name="set_all_stream_rate" args="call --wait /mavros/set_stream_rate 0 0 1"/>
  <node pkg="rosservice" type="rosservice" name="set_odom_stream_rate" args="call --wait /mavros/set_stream_rate 3 $(arg stream_rate) 1" launch-prefix="bash -c 'sleep 1; $0 $@' "/>
  <node pkg="rosservice" type="rosservice" name="set_rc_stream_rate" args="call --wait /mavros/set_stream_rate 6 $(arg stream_rate) 1" launch-prefix="bash -c 'sleep 1; $0 $@' "/>
  <node pkg="rosservice" type="rosservice" name="set_position_stream_rate" args="call --wait /mavros/set_stream_rate 10 $(arg stream_rate) 1" launch-prefix="bash -c 'sleep 1; $0 $@' "/>


  <node pkg="data_reformater" type="data_reformater" name="data_reformater" required="true" clear_params="true" output="screen" respawn="false"/>


  <node if="$(arg do_logging)" name="dfti" pkg="dfti2" type="dfti" output="screen" launch-prefix="bash -c 'sleep $(arg logging_start_delay); $0 $@' "/>
</launch>
